### PR TITLE
New CLI with new filter order. First version of space removing filter.

### PR DESCRIPTION
**Pull request by SunBear1**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
-        for word in badwords:
-            run_filters(word=word)
+    option = input("1) Wczytaj tekst z pliku\n2) Wczytaj tekst z konsoli\n")
+    if option == "1":
+        text_to_filter = load_input_from_file(path=INPUT_FILE_PATH)
+    elif option == "2":
+        text_to_filter = input("Wprowadź tekst:\n")
     else:
-        run_filters(word=user_input)
+        option = input("Nieprawidłowa opcja\n1) Wczytaj tekst z pliku\n2) Wczytaj tekst z konsoli\n")
```
Comment (jereml99) [inline src/main.py:17]: Don't we need wail here?
```suggestion
good_input = false
while good_input:
    option = input("1) Wczytaj tekst z pliku\n2) Wczytaj tekst z konsoli\n")
    if option == "1":
        text_to_filter = load_input_from_file(path=INPUT_FILE_PATH)
        good_input = true
    elif option == "2":
        text_to_filter = input("Wprowadź tekst:\n")
        good_input = true
    else:
        run_filters(word=user_input)
        option = input("Nieprawidłowa opcja\n1) Wczytaj tekst z pliku\n2) Wczytaj tekst z konsoli\n")
```

Comment (jereml99) [inline src/main.py:17]: Don't we need wail here?
```suggestion
good_input = false
while good_input:
    option = input("1) Wczytaj tekst z pliku\n2) Wczytaj tekst z konsoli\n")
    if option == "1":
        text_to_filter = load_input_from_file(path=INPUT_FILE_PATH)
        good_input = true
    elif option == "2":
        text_to_filter = input("Wprowadź tekst:\n")
        good_input = true
    else:
        run_filters(word=user_input)
        option = input("Nieprawidłowa opcja\n1) Wczytaj tekst z pliku\n2) Wczytaj tekst z konsoli\n")
```

File: src/main.py [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
+        no_spaces = remove_spaces(first_word=text_to_filter[i], second_word=text_to_filter[i + 1])
+        no_repeats = remove_repeats(word=no_spaces)
+        for word in charswap_filter(word=no_repeats):
+            no_endings = remove_endings(word)
+            if no_endings in SourceDictionaries.RAW_BAD_WORDS:
+                text_to_filter[i] = str(text_to_filter[i] + text_to_filter[i + 1])
+                text_to_filter[i] = text_to_filter[i].replace(text_to_filter[i], ("*" * len(text_to_filter[i])))
+                text_to_filter.pop(i + 1)
+        i += 1
+    print(text_to_filter)
```
Comment (jereml99) [inline src/main.py:47]: Maybe move it to separate function

**Review by jereml99 — CHANGES_REQUESTED** [review]

File: src/main.py (hunk 1 above)
Comment (SunBear1) [reply src/main.py:17]: The line 22 works as one, but I think your solution is better

Comment (FreddieHorn) [reply src/main.py]: you can change good_input to bad_input since its more understandable

File: src/main.py (hunk 2 above)
Comment (FreddieHorn) [reply src/main.py]: Or just make lines 43 - 45 a seperate function so it fits the ones invoked in lines 38, 39, 40 and 41

**Review by jereml99 — APPROVED** [review]

**Review by FreddieHorn — APPROVED** [review]

**Pull request merged**